specs/ExtendedProcessFailed.md

```diff
@@ -0,0 +1,671 @@
+library WebView2
+{
+// ...
+
+/// Specifies the process failure type used in the
+/// `ICoreWebView2ProcessFailedEventHandler` interface. The values in this enum
+/// make reference to the process kinds in the Chromium architecture. For more
+/// information about what these processes are and what they do, see
+/// [Browser Architecture - Inside look at modern web browser](https://developers.google.com/web/updates/2018/09/inside-browser-part1)
+[v1_enum]
+typedef enum COREWEBVIEW2_PROCESS_FAILED_KIND {
+  // Existing stable values.
+
+  // Note: docs for this enum value remain unchanged.
+  /// Indicates that the browser process ended unexpectedly.  The WebView
+  /// automatically moves to the Closed state.  The app has to recreate a new
+  /// WebView to recover from this failure.
+  COREWEBVIEW2_PROCESS_FAILED_KIND_BROWSER_PROCESS_EXITED,
+
+  /// Indicates that the main frame's render process ended unexpectedly.  A new
+  /// render process is created automatically and navigated to an error page.
+  /// You can use the Reload method to try to reload the page that failed.
+  COREWEBVIEW2_PROCESS_FAILED_KIND_RENDER_PROCESS_EXITED,
+
+  /// Indicates that the main frame's render process is unresponsive.
+  COREWEBVIEW2_PROCESS_FAILED_KIND_RENDER_PROCESS_UNRESPONSIVE,
+
+  // New values.
+  
+  /// Indicates that a frame-only render process ended unexpectedly. The process
+  /// exit does not impact the top-level document, only a subset of the
+  /// subframes within it. The content in these frames is replaced with an error
+  /// page in the frame.
+  COREWEBVIEW2_PROCESS_FAILED_KIND_FRAME_RENDER_PROCESS_EXITED,
+
+  /// Indicates that a utility process ended unexpectedly.
+  COREWEBVIEW2_PROCESS_FAILED_KIND_UTILITY_PROCESS_EXITED,
+
+  /// Indicates that a sandbox helper process ended unexpectedly.
+  COREWEBVIEW2_PROCESS_FAILED_KIND_SANDBOX_HELPER_PROCESS_EXITED,
+
+  /// Indicates that the GPU process ended unexpectedly.
+  COREWEBVIEW2_PROCESS_FAILED_KIND_GPU_PROCESS_EXITED,
+
+  /// Indicates that a PPAPI plugin process ended unexpectedly.
+  COREWEBVIEW2_PROCESS_FAILED_KIND_PPAPI_PLUGIN_PROCESS_EXITED,
+
+  /// Indicates that a PPAPI plugin broker process ended unexpectedly.
+  COREWEBVIEW2_PROCESS_FAILED_KIND_PPAPI_BROKER_PROCESS_EXITED,
+
+  /// Indicates that a process of unspecified kind ended unexpectedly.
+  COREWEBVIEW2_PROCESS_FAILED_KIND_UNKNOWN_PROCESS_EXITED,
+} COREWEBVIEW2_PROCESS_FAILED_KIND;
+
+
+/// Specifies the process failure reason used in the
+/// `ICoreWebView2ProcessFailedEventHandler` interface.
+[v1_enum]
+typedef enum COREWEBVIEW2_PROCESS_FAILED_REASON {
+  /// An unexpected process failure occurred.
+  COREWEBVIEW2_PROCESS_FAILED_REASON_UNEXPECTED,
+
+  /// The process became unresponsive.
+  /// This only applies to the main frame's render process.
+  COREWEBVIEW2_PROCESS_FAILED_REASON_UNRESPONSIVE,
+
+  /// The process was killed. For example, from Task Manager.
+  COREWEBVIEW2_PROCESS_FAILED_REASON_KILLED,
```

Comment:
Perhaps "TERMINATED" or "EXTERNAL" would avoid the scary word 'KILLED".

Reply:
I like "Terminated"